src/app/components/content-viewer/content-viewer.component.ts

```diff
@@ -52,7 +54,6 @@ export class ContentViewerComponent implements OnDestroy {
   private componentRef: ComponentRef<{}>;
 
   constructor(
-    private parser: GenericParserService,
   ) {
     this.context$.subscribe();
```

Comment:
no need for this subscription we already manage the subscription in the template `<ng-container *ngIf="context$ | async as context">
`

Reply:
It doesn't work without, and I don't understand why

Reply:
It does work only when context$ is a BehaviorSubject. I've switched type to that.